util/config/compileline

```diff
@@ -1,154 +1,201 @@
-    print " --compile  print out the command to compile a file including\n";
-    print "            Chapel runtime headers\n";
-    print " --includes-and-defines print out any -D and -I options to\n";
-    print "                        include the Chapel runtime\n";
-    print " --libraries print out any -L and -l options to\n";
-    print "             link with the Chapel runtime\n";
-    print " --main.o print out the path to the main.o file\n";
-    print " --clang print out the path to clang and clang++\n";
-    print " --clang-sysroot-arguments print out any saved clang arguments\n";
-    print "                           that specify the system root\n";
-  }
-}
+  ret = ""
+  output = utils.run_command(cmd)
+  lines = output.splitlines()
+  for line in lines:
+    if not line.startswith("make"):
+      ret += line
+
+  return ret
+
+def myrun2(cmd, arg):
+
+  return myrun(cmd + [arg])
+
+def mysystem(cmd):
+
+  s = myrun(cmd)
+  sys.stdout.write(s)
+
+def mysystem2(cmd, arg):
+
+  mysystem(cmd + [arg])
+
+def _main():
+
+  global chpl_home_dir
+  check_file = os.path.relpath(__file__, chpl_home_dir)
+  check_path = os.path.join(chpl_home_dir, check_file)
+  if not os.path.isfile(check_path):
+    sys.stderr.write("Warning: check {0} not found\n".format(check_path));
+
+  if "CHPL_HOME" in os.environ:
+    if os.path.abspath(os.environ["CHPL_HOME"]) != chpl_home_dir:
+      # to be sure, check that the inode numbers of our check file match
+      env_check = os.path.join(os.environ["CHPL_HOME"], check_file)
+      rel_check = os.path.join(chpl_home_dir, check_file)
+      if os.path.samefile(env_check, rel_check):
+        # No warning, it's OK, they are the same file.
+        pass
+      else:
+        sys.stderr.write("Warning: check {0} not found\n".format(check_path));
+        sys.stderr.write("Warning: Mismatched CHPL_HOME; got {0} but expected {1}\n".format(os.path.abspath(os.environ["CHPL_HOME"]), chpl_home_dir));
+    chpl_home_dir = os.environ["CHPL_HOME"] # use enviro var spelling of it
+  else:
+    os.environ["CHPL_HOME"] = chpl_home_dir
+
+  os.environ["CHPL_MAKE_HOME"] = chpl_home_dir;
+  if "CHPL_RUNTIME_LIB" in os.environ:
+    os.environ["CHPL_MAKE_RUNTIME_LIB"] = os.environ["CHPL_RUNTIME_LIB"]
+
+  if "CHPL_RUNTIME_INCL" in os.environ:
+    os.environ["CHPL_MAKE_RUNTIME_INCL"] = os.environ["CHPL_RUNTIME_INCL"]
+
+  if "CHPL_THIRD_PARTY" in os.environ:
+    os.environ["CHPL_MAKE_THIRD_PARTY"] = os.environ["CHPL_THIRD_PARTY"]
+
+  make = chpl_make.get()
+
+  orig_make = make
+  # Do not print directory changes.
+  make = [make, "--no-print-directory"]
+
+  # Make reasonable defaults for environment settings
+  os.environ["COMP_GEN_WARN"] = "0";
+  os.environ["COMP_GEN_DEBUG"] = "0";
+  os.environ["COMP_GEN_OPT"] = "0";
+  os.environ["COMP_GEN_SPECIALIZE"] = "0";
+  os.environ["COMP_GEN_IEEE_FLOAT"] = "1";
+
+  parser = optparse.OptionParser(usage="usage: %prog [ENVVAR=VALUE] [flags]")
```

Comment:
It would be nice to consolidate the arg parsing into a function that just returns `parser.parse_args()`